### ISSUE TITLE
`paste_linter()` doesn't work as expected with pipes

### ISSUE THREAD

**Issue by IndrajeetPatil** [r-lib/lintr#2571]
``` r
library(lintr)

lint(text = 'paste0(raw, collapse = "")', linters = paste_linter())
#> <text>:1:1: warning: [paste_linter] Use paste(), not paste0(), to collapse a character vector when sep= is not used.
#> paste0(raw, collapse = "")
#> ^~~~~~~~~~~~~~~~~~~~~~~~~~
lint(text = 'raw %>% paste0(collapse = "")', linters = paste_linter())
```

<sup>Created on 2024-05-05 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

**Comment by MichaelChirico** [comment]
One more too:

```r
lint(text = 'x |> paste0(y, collapse = "")', linters=paste_linter())
# <text>:1:6: warning: [paste_linter] Use paste(), not paste0(), to collapse a character vector when sep= is not used.
# x |> paste0(y, collapse = "")
#      ^~~~~~~~~~~~~~~~~~~~~~~~
```
